File: ios/Lumos.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Lumos, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(lumos)

RCT_EXTERN_METHOD(nox)

@end

File: ios/Lumos.swift
import  AVFoundation

@objc(Lumos)
class Lumos: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        print("Multiply is called natively")
        resolve(a*b)
    }
    
    @objc
    func lumos() {
        toggleTorch(on: true)
    }
    
    @objc
    func nox() {
           toggleTorch(on: false)
       }
    
    private func toggleTorch(on: Bool) {
        guard let device = AVCaptureDevice.default(for: AVMediaType.video) else { return }
        guard device.hasTorch else { print("Torch isn't available"); return }

        do {
            try device.lockForConfiguration()
            device.torchMode = on ? .on : .off
            if on { try device.setTorchModeOn(level: AVCaptureDevice.maxAvailableTorchLevel.significand) }
            device.unlockForConfiguration()
        } catch {
            print("Torch can't be used")
        }
    }
}
